server/game/cards/characters/02/sergregorclegane.js

```diff
@@ -0,0 +1,50 @@
+const DrawCard = require('../../../drawcard.js');
+
+class SerGregorClegane extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCardDiscarded: (event, challenge, card) => challenge.winner === this.controller && card === this
+            },
+            handler: context => {
+                var discardedCard = context.event.params[2];
```

Comment:
Unfortunately, this will be the card that did the pillaging (Gregor in this case), not the card that got discarded. The `onPillage` event raised in the `ChallengeFlow` class will need to be changed to also pass the card that got discarded.